File: StayBookingFrontend/lib/model/login_request_dto.dart
import 'package:stay_booking_frontend/model/app_role.dart';

class LoginRequestDto {
  LoginRequestDto({
    required this.email,
    required this.password,
    required this.role,
  });

  final String email;
  final String password;
  final AppRole role;

  Map<String, dynamic> toJson() {
    return {
      'email': email,
      'password': password,
      'role': role.apiValue,
    };
  }
}

File: StayBooking/StayBooking/src/main/java/com/dto/LoginRequestDto.java
package com.dto;

import com.enums.Role;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

@Data
public class LoginRequestDto {
    @NotBlank(message = "email is required")
    private String email;

    @NotBlank(message = "password is required")
    private String password;

    @NotNull(message = "role is required")
    private Role role;
}
